src/PrestaShopBundle/Form/Admin/Type/ProductSearchType.php

```diff
@@ -0,0 +1,115 @@
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type;
+
+use PrestaShopBundle\Form\Admin\Sell\Product\SearchedProductItemType;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Initiates input with ability to search for any type of product
+ */
+class ProductSearchType extends TranslatorAwareType
+{
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var string
+     */
+    private $languageIsoCode;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        RouterInterface $router,
+        string $employeeIsoCode
+    ) {
+        parent::__construct($translator, $locales);
+        $this->router = $router;
+        $this->languageIsoCode = $employeeIsoCode;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        parent::configureOptions($resolver);
+
+        $refLabel = $this->trans('Ref: %s', 'Admin.Catalog.Feature');
+        $combinationsUrl = $this->router->generate('admin_products_v2_search_combinations', [
+            'languageCode' => $this->languageIsoCode,
+            'query' => '__QUERY__',
+        ]);
+        $productsUrl = $this->router->generate('admin_products_v2_search_associations', [
+            'languageCode' => $this->languageIsoCode,
+            'query' => '__QUERY__',
+        ]);
+        $resolver->setDefault('search_combinations', true);
+        $resolver->setDefaults([
+            'required' => false,
+            'label' => false,
+            'placeholder' => $this->trans('Search product', 'Admin.Catalog.Help'),
+            'min_length' => 3,
+            'limit' => 1,
+            'identifier_field' => static function (Options $options): string {
+                return $options->offsetGet('search_combinations') === true ? 'unique_identifier' : 'id';
+            },
+            'entry_type' => static function (Options $options): string {
+                return $options->offsetGet('search_combinations') === true ? SearchedProductItemType::class : EntityItemType::class;
```

Comment:
Since `SearchedProductItemType` is only used when searching for combinations, shouldn't it be named `SearchedCombinationItemType` instead?

Reply:
It is hard to make the naming right as probably the endpoints are the problem.
The endpoint searches for all types of product including combinations, so I tend to think it would be even more confusing if i name it combination.

Reply:
maybe I should only use the SearchedProductItem type everywhere, and just ignore the combination_id (which would be null) where its not combinational search?